src/edit_view.rs

```diff
@@ -685,10 +683,28 @@ impl EditView {
         let pango_ctx = self.view_item.get_pango_ctx();
         pango_ctx.set_font_description(&self.edit_font.font_desc);
 
-        // Draw editing background
-        set_source_color(cr, theme.background);
-        cr.rectangle(0.0, 0.0, f64::from(da_width), f64::from(da_height));
-        cr.fill();
+        // Draw a line at x chars
+        if get_draw_right_margin() {
+            let until_margin_width = self.edit_font.font_width * get_column_right_margin() as f64;
+            // Draw editing background
+            set_source_color(cr, theme.background);
+            cr.rectangle(0.0, 0.0, until_margin_width, f64::from(da_height));
+            cr.fill();
```

Comment:
We use whatever Xi tells us the background is here (this is the darker part in the screenshot)